TCAT/Controllers/RouteDetailContentViewController+Extensions.swift

```diff
@@ -0,0 +1,191 @@
+        didUpdateLocation()
+    }
+
+    func locationManager(_ manager: CLLocationManager, didFailWithError error: Swift.Error) {
+        print("RouteDetailVC CLLocationManager didFailWithError: \(error)")
+        didUpdateLocation()
+    }
+}
+
+// MARK: Google Map View Delegate Functions
+extension RouteDetailContentViewController: GMSMapViewDelegate {
+
+    func mapView(_ mapView: GMSMapView, didTap marker: GMSMarker) -> Bool {
+        guard let coordinates = getUserData(for: marker, key: Constants.BusUserData.actualCoordinates) as? CLLocationCoordinate2D
+            else { return true }
+        let update = GMSCameraUpdate.setTarget(coordinates)
+        mapView.animate(with: update)
+
+        return true
+    }
+
+    func updateUserData(for marker: GMSMarker, with values: [String: Any]) {
+        guard var userData = marker.userData as? [String: Any] else {
+            marker.userData = values
+            return
+        }
+        values.forEach { (key, value) in
+            userData[key] = value
+        }
+        marker.userData = userData
+    }
+
+    func getUserData(for marker: GMSMarker, key: String) -> Any? {
+        guard let userData = marker.userData as? [String: Any] else { return nil }
+        return userData[key]
+    }
+
+    func mapView(_ mapView: GMSMapView, didChange position: GMSCameraPosition) {
+        buses.forEach { bus in
+            let bearingView = UIImageView(image: #imageLiteral(resourceName: "indicator"))
+            bearingView.frame.size = CGSize(width: bearingView.frame.width / 2, height: bearingView.frame.height / 2)
+            bearingView.tag = increaseTapTargetTag
+
+            if let existingIndicator = busIndicators.first(where: {
+                let markerID = getUserData(for: $0, key: Constants.BusUserData.vehicleID) as? Int
+                let busID = getUserData(for: bus, key: Constants.BusUserData.vehicleID) as? Int
+                return markerID == busID
+            }) { // Update Indicator
+                if let placement = calculatePlacement(position: bus.position, view: bearingView) {
+
+                    // existingIndicator.map = nil // Uncomment to avoid animation
+                    existingIndicator.position = placement
+                    existingIndicator.rotation = calculateBearing(from: placement, to: bus.position)
+
+                    updateUserData(for: existingIndicator, with: [
+                        Constants.BusUserData.actualCoordinates: bus.position,
+                        Constants.BusUserData.indicatorCoordinates: placement
+                        ])
+
+                    existingIndicator.appearAnimation = .none
+                    // existingIndicator.map = mapView // Uncomment to avoid animation
+                } else {
+                    existingIndicator.map = nil
+                    busIndicators.remove(at: busIndicators.firstIndex(of: existingIndicator)!)
+                }
+            } else { // Create Indicator
```

Comment:
might be easier to read if you add a return after line 150 and then remove this `else` block